dev/run

```diff
@@ -505,6 +517,119 @@ def boot_nouveau(ctx):
     )
 
 
+CLOUSEAU_DIR = "clouseau"
+JAVA_VERSION_RE = re.compile('"(\d+\.\d+).*"')
+
+
+def get_java_version(java):
+    output = sp.check_output([java, "-version"], stderr=sp.STDOUT).decode("utf-8")
+    if output:
+        matches = JAVA_VERSION_RE.search(output)
+        if matches:
+            return float(matches.groups()[0])
+
+
+class ClouseauError(Exception):
+    def __init__(self, message):
+        self.message = message
+        super().__init__(self.message)
+
+
+@log("Start Clouseau node clouseau{idx}")
+def boot_clouseau(ctx, idx):
+    if not os.path.isdir(CLOUSEAU_DIR):
+        raise ClouseauError(
+            "Clouseau deployment cannot be found, please run `./configure with --enable-clouseau`"
+        )
+
+    clouseau_jdk_home = os.environ.get("CLOUSEAU_JAVA_HOME") or os.environ.get(
+        "JAVA_HOME"
+    )
+    java = clouseau_jdk_home + "/bin/java" if clouseau_jdk_home else "java"
+    java_version = get_java_version(java)
+
+    if not java_version:
+        print(
+            "Warning: Java version could not be determined, Clouseau may not be able to run"
+        )
```

Comment:
Shall we use `raise ClouseauError()` here as well?

Tried with Java openjdk-17, get the following error
```bash
[ * ] Start Clouseau node clouseau1 ... Warning: Java version could not be determined, Clouseau may not be able to run
failed: '<' not supported between instances of 'NoneType' and 'float'
Traceback (most recent call last):
  File "/Users/jiahuili/src/pgj/./dev/run", line 1064, in <module>
    main()
  File "/Users/jiahuili/src/pgj/./dev/run", line 94, in main
    startup(ctx)
  File "/Users/jiahuili/src/pgj/./dev/run", line 711, in startup
    boot_nodes(ctx)
  File "/Users/jiahuili/src/pgj/./dev/run", line 757, in boot_nodes
    clouseau_proc = maybe_boot_clouseau(ctx, idx)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jiahuili/src/pgj/./dev/run", line 611, in maybe_boot_clouseau
    return boot_clouseau(ctx, idx)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jiahuili/src/pgj/./dev/run", line 73, in wrapper
    res = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jiahuili/src/pgj/./dev/run", line 556, in boot_clouseau
    if java_version < 1.7 or java_version > 1.8:
       ^^^^^^^^^^^^^^^^^^
TypeError: '<' not supported between instances of 'NoneType' and 'float'
```

Reply:
This is a not a real error: it still might be possible to run Clouseau, but there are no guarantees for that.  However, the `TypeError` above you included should not be there, thanks for reporting.

Reply:
No `TypeError` on my end, +1, and thanks for the improvement!